stellar-contract-env-common/src/raw_val.rs

```diff
@@ -61,7 +61,18 @@ impl AsMut<RawVal> for RawVal {
 // away, the same way `()` would, while remaining a separate type to allow
 // conversion to a more-structured error code at a higher level.
 #[derive(Debug)]
-pub struct ConversionError;
+pub struct ConversionError<F, T>;
+
+impl<F, T> core::fmt::Display for ConversionError<F, T> {
+    fn fmt(&self, f: &mut core::fmt::Formatter<'_>) -> core::fmt::Result {
+        write!(
+            f,
+            "Conversion from {} to {} failed",
+            std::any::type_name::<F>(),
+            std::any::type_name::<T>()
+        )
+    }
+}
```

Comment:
The idea is to pass in types `F` and `T` (denoting the data *from* and *to* for the conversion) into ConversionError but with no actual data. Then use the `std::any::type_name::<T>` to print out a string representation of the type `T`.  Implement `core::fmt::Display` or `Debug` for `ConversionError` so that it can get back a string. 

The code as presented complains about unused parameter `F` and `T` in `pub struct ConversionError<F, T>`. 
I tried `core::fmt:Debug` as well and it wouldn't work either due to a different error. 
I tried putting in `PhantomData` inside `ConversionError`, i.e. 

`pub struct ConversionError<F, T> {
    _f: PhantomData<F>,
    _t: PhantomData<T>,
}`

But couldn't figure out how to make the rest of it work. I will try a bit more tomorrow. 

@leighmcculloch Does this align with what you had in mind?  
Suggestions are welcome. cc @graydon @jonjove

Reply:
Thanks @leighmcculloch , I like your suggestion of passing in a `RawVal` and leverage the printing feature of it. 

I made the changes you suggested and got it basically working, see updated description. 

I'm not sure if I fully understand what you meant by:

> Will having generics in the error make it difficult to wrap this error type?

But I think I addressed it in my code. See bullet point #2 in the description.